server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -156,12 +156,23 @@ private void getMultipleReposSnapshotInfo(
                     .collect(Collectors.toMap(Tuple::v1, Tuple::v2));
                 final SnapshotsInRepo snInfos = sortSnapshots(allSnapshots, sortBy, after, size, order);
                 final List<SnapshotInfo> snapshotInfos = snInfos.snapshotInfos;
+                final int remaining;
+                final boolean hasMore = snInfos.hasMore || responses.stream().anyMatch(r -> r.v2() != null && r.v2().hasMore);
```

Comment:
The logic for all this counting could be made for nicer obviously, but for now I went with the shortest route to getting this implemented on top of the existing functionality as we will be changing how this works under the hood soon anyway.

Reply:
Right, dropped `hasMore` completely now :)